R/datasets.R

```diff
@@ -0,0 +1,627 @@
+#'
+#' @description
+#' Filter Tabular Dataset between a specified start and end time.
+#'
+#' @param dataset The Tabular Dataset object
+#' @param start_time The lower bound for filtering data.
+#' @param end_time The upper bound for filtering data.
+#' @param include_boundary Boolean indicating if the row associated with the
+#' boundary time (`start_time` and `end_time`) should be included.
+#' @return The filtered Tabular Dataset
+#' @export
+#' @md
+filter_dataset_between_time <- function(dataset, start_time, end_time,
+                                        include_boundary = FALSE) {
+  dataset$time_between(start_time, end_time, include_boundary)
+}
+
+#' Filter Tabular Dataset to contain only the specified duration (amount) of recent data.
+#'
+#' @description
+#' Filter Tabular Dataset to contain only the specified duration (amount) of recent data.
+#'
+#' @param dataset The Tabular Dataset object
+#' @param time_delta The duration (amount) of recent data to retrieve.
+#' @param include_boundary Boolean indicating if the row associated with the
+#' boundary time (`time_delta`) should be included.
+#' @return The filtered Tabular Dataset
+#' @export
+#' @md
+filter_dataset_from_recent_time <- function(dataset, time_delta,
+                                            include_boundary = FALSE) {
+  dataset$time_recent(time_delta, include_boundary)
+}
+
+#' Define timestamp columns for the dataset.
+#'
+#' @description
+#' Define timestamp columns for the dataset.
+#' The method defines columns to be used as timestamps. Timestamp columns on a dataset
+#' make it possible to treat the data as time-series data and enable additional capabilities.
+#' When a dataset has both `fine_grain_timestamp` and `coarse_grain_timestamp defined`
+#' specified, the two columns should represent the same timeline.
+#'
+#' @param dataset The Tabular Dataset object.
+#' @param fine_grain_timestamp The name of column as fine grain timestamp. Use None to clear it.
+#' @param coarse_grain_timestamp The name of column coarse grain timestamp (optional).
+#' The default is None.
+#' @param validate Indicates whether to validate if specified columns exist in dataset.
+#' The default is False. Validation requires that the data source is accessible
+#' from the current compute.
+#' @return The Tabular Dataset with timestamp columns defined.
+#' @export
+#' @md
+define_timestamp_columns_for_dataset <- function(dataset, fine_grain_timestamp,
+                                                 coarse_grain_timestamp = NULL,
+                                                 validate = FALSE) {
+  dataset$with_timestamp_columns(fine_grain_timestamp, coarse_grain_timestamp,
+                                 validate)
+}
+
+#' Load all records from the dataset into a pandas DataFrame.
+#'
+#' @description
+#' Load all records from the dataset into a pandas DataFrame.
+#'
+#' @param dataset The Tabular Dataset object.
+#' @return A pandas DataFrame.
```

Comment:
Should return an R data frame not a pandas dataframe

Reply:
Does reticulate automatically handle this conversion right?

Reply:
i think we need to explicitly call reticulate API for the conversion

Reply:
Pandas dataframes are automatically converted to R dataframes: https://rstudio.github.io/reticulate/articles/calling_python.html